### PR TITLE
PRO-16890 : Enforce required flags for provar:metadatacache

### PR DESCRIPTION
**Pull request by MayankNehru**
The command started executing even when required flags were not given. Changed that to give error in typescript itself. Also made changes to make sure that the flags work correctly in tandem.

### CODE REVIEW

File: src/commands/provar/metadatacache.ts [hunk 1]
```diff
@@ -81,6 +81,24 @@ export default class MetadataCache extends SfdxCommand {
...
+        if (
+            !metadataLevel || !cachePath
+        ) {
+            this.ux.error(
+                "ERROR running provar:metadatacache :  Please specify a cachepath and metadatalevel"
+            );
+            return {};
+        }
+
+        if (
```
Comment (riteshnahar) [inline src/commands/provar/metadatacache.ts]: Change it to:
`if (!["Reload", "Refresh", "Reuse"].includes(metadataLevel))`

File: src/commands/provar/metadatacache.ts [hunk 2]
```diff
@@ -117,26 +135,26 @@ export default class MetadataCache extends SfdxCommand {
...
-        if (userInfo == null) {
+        if (userInfo == null && !connections) {
             this.ux.error(
                 '[ERROR] No valid user org found to download metadata. Terminating command.'
             );
             return {};
         }
 
-        const userInfoString = provarDxUtils.prepareRawProperties(
+        const userInfoString = connections && userInfo == null ? "NA" : provarDxUtils.prepareRawProperties(
```
Comment (riteshnahar) [inline src/commands/provar/metadatacache.ts]: `userInfo === null`

**Review by riteshnahar — CHANGES_REQUESTED** [review]
Please do necessary changes in `CHANGELOG.md` file as well

**Review by riteshnahar — APPROVED** [review]

File: src/commands/provar/metadatacache.ts [hunk 3]
```diff
@@ -81,6 +81,22 @@ export default class MetadataCache extends SfdxCommand {
...
+        if (
+            !metadataLevel || !cachePath
+        ) {
+            this.ux.error(
+                "ERROR running provar:metadatacache :  Please specify a cachepath and metadatalevel"
+            );
+            return {};
+        }
+
+        (!["Reload", "Refresh", "Reuse"].includes(metadataLevel)) {
```
Comment (simoncarter-provar) [inline src/commands/provar/metadatacache.ts]: Should this be an if statement?

`if (!["Reload", "Refresh", "Reuse"].includes(metadataLevel)) {`

**Review by simoncarter-provar — CHANGES_REQUESTED** [review]

**Review by simoncarter-provar — APPROVED** [review]

**Pull request merged**